lib/spack/spack/test/cmd/buildcache.py

```diff
@@ -848,3 +850,192 @@ def test_buildcache_prune_orphaned_manifest(tmp_path, mutable_database, mock_gnu
+        "--name",
+        "test_view",
+        "my-mirror",
+        str(empty_lockfile),
+    ]
+    out = buildcache(*command_args)
+    assert "No specs found for view, creating an empty index" in out
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    assert not hashes_in_view
+
+
+@pytest.mark.parametrize("sources", ("hash", "specfile", "env", "lockfile"))
+def test_buildcache_create_view(
+    tmp_path, mutable_config, mutable_database, mutable_mock_env_path, sources
+):
+    mirror_directory = str(tmp_path)
+    mirror("add", "--unsigned", "my-mirror", mirror_directory)
+
+    # Push all of the mpileaks packages to the cache
+    mpileaks_specs = mutable_database.query_local("mpileaks")
+    for s in mpileaks_specs:
+        buildcache("push", "--update-index", "my-mirror", s.format("{/hash}"))
+
+    # Grab all of the hashes for each mpileaks spec
+    mpileaks_0_hashes = [s.dag_hash() for s in mpileaks_specs[0].traverse()]
+    mpileaks_1_hashes = [s.dag_hash() for s in mpileaks_specs[1].traverse()]
+    mpileaks_2_hashes = [s.dag_hash() for s in mpileaks_specs[2].traverse()]
+
+    # Make sure the view doesn't exist yet
+    with pytest.raises(spack.binary_distribution.FetchIndexError):
+        hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+
+    # Create a view using a parametrized source that contains one of the mpileaks
+    with globals()[f"spec_{sources}"](mpileaks_specs[0], tmp_path / "source_stage") as source_args:
+        command_args = ["create-view", "--force", "--name", "test_view", "my-mirror"]
+        if source_args:
+            command_args.extend(source_args)
+        buildcache(*command_args)
+
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    for h in mpileaks_0_hashes:
+        assert h in hashes_in_view
+
+    # Test failure due to existing index
+
+    # Test append
+    with globals()[f"spec_{sources}"](mpileaks_specs[1], tmp_path / "source_stage") as source_args:
+        command_args = ["update-view", "--append", "--name", "test_view", "my-mirror"]
+        if source_args:
+            command_args.extend(source_args)
+        buildcache(*command_args)
+
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    for h in mpileaks_0_hashes + mpileaks_1_hashes:
+        assert h in hashes_in_view
+
+    # Test fail to overwrite
+    expect = "Index already exists. To overwrite or update pass --force or --append respectively"
+    with pytest.raises(spack.error.SpackError, match=expect):
+        command_args = [
+            "update-view",
+            "--name",
+            "test_view",
+            "my-mirror",
+            mpileaks_specs[2].format("{/hash}"),
+        ]
+        buildcache(*command_args)
+
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    for h in mpileaks_0_hashes + mpileaks_1_hashes:
+        assert h in hashes_in_view
+
+    # Test fail to overwrite
```

Comment:
Seems like this comment is stale and you are testing intersection of the view vs the full cache? Would be clearer as a separate test.